src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1301,7 +1301,7 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
                     //
                     // Note that it is safe to devirtualize in the following cases, since a servicing event cannot later modify it
                     //  1) Callvirt on a virtual final method of a value type - since value types are sealed types as per ECMA spec
-                    devirt = (constrainedType ?? targetMethod.OwningType).IsValueType || targetMethod.IsInternalCall;
+                    devirt = targetMethod.OwningType.IsValueType || targetMethod.IsInternalCall;
```

Comment:
Btw, this code is still different from what crossgen does - the comment above ends at `1)`, but the VM side special cases 3 things - this seems to be missing an optimization around `Delegate.Invoke` - we probably generate less efficient code for that as a result.

Reply:
I added the two other cases (delegates and intrinsics) so we line with up with CG1